File: cmstestsuite/code/half-correct-fileio.cpp
#include <fstream>

int main() {
    int n;
    std::ifstream in("input.txt");
    std::ofstream out("output.txt");
    in >> n;
    out << "correct " << (n % 2 ? n : 0) << std::endl;
    return 0;
}

File: cmstestsuite/code/half-correct-fileio.php
<?php
$fin = fopen("input.txt", "r");
$n = intval(fgets($fin));
$fout = fopen("output.txt", "w");
if ($n % 2 == 0)
    fwrite($fout, "correct 0\n");
else
    fwrite($fout, "correct $n\n");
?>
